core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -552,10 +552,14 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
     // locked. This could improve performance. Further investigation is needed.
 
     // Fill in the ancestor directories that were missing.
+    // NOTE, we set the permission mode of missing ancestor directories to be the default rather
+    // than inheriting from the option of the final file to create, because it may not have
+    // "execute" permission.
     CreateDirectoryOptions missingDirOptions = CreateDirectoryOptions.defaults()
         .setMountPoint(false)
         .setPersisted(options.isPersisted())
-        .setPermission(options.getPermission());
+        .setPermission(options.getPermission())
```

Comment:
if you implement my suggestion to get rid of mDefaultMode (see below), then we will not need to call setPermission or setDefaultMode here